pandas/io/pytables.py

```diff
@@ -3036,7 +3037,7 @@ def write_array_empty(self, key: str, value: ArrayLike):
         node._v_attrs.shape = value.shape
 
     def write_array(
-        self, key: str, obj: DataFrame | Series, items: Index | None = None
+        self, key: str, obj: AnyArrayLike, items: Index | None = None
```

Comment:
Not entirely sure on this one, but judging from the code and usage it looks like this method cannot accept a DataFrame

Reply:
Can't comment on `DataFrame` @simonjayhawkins 

If `DataFrame` cannot be accepted in practice, why not simplify the type to just `Series`?

Reply:
I think that for example [here](https://github.com/pandas-dev/pandas/blob/main/pandas/io/pytables.py#L2934) this method is called with an ndarray